modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts

```diff
@@ -83,15 +83,70 @@ export interface SiteConfig<Config> {
     config: Config;
 }
 
-export interface AssetUrlParams {
+export interface Cookie {
+    value: string;
+    path?: string;
+    domain?: string;
+    maxAge?: number;
+    secure?: boolean;
+    httpOnly?: boolean;
+    sameSite?: 'lax' | 'strict' | 'none' | '';
+}
+
+type LiteralUnion<T extends U, U = string> = T | (U & Record<never, never>);
+
+export type RequestParams = Record<string, string | string[] | undefined>;
+
+export interface Request<Params extends RequestParams = RequestParams,
```

Comment:
Did you consider removing (all or some of) the type parameters like I suggested earlier?